### ISSUE TITLE
Convert .dds files to .itp?

### ISSUE THREAD

**Issue by cloudyst** [Coxxs/itpcnv-dana#1]
hello,

sorry if this doesn't go here, but it's the only way I can see to contact you.
I'm trying to translate the ys8 game into my language (spanish) and of course, there is a lot of text that is image. This application works great, I can open the image files to be able to modify them, but how do I go back to the previous step to make them work in the game?

Is it necessary to create something else with this application? could you make it? 
im using the pc version.

thank you very much and sorry for the inconvenience

**Comment by Coxxs** [comment]
This program does following things (source code is available in the repo):
1. [Decompress the .itp file using the algorithm obtained in the original binary.](https://github.com/Coxxs/itpcnv-dana/blob/master/index.js#L66
) (The compression algorithm might not present in the original binary, you can try [disable compression](https://github.com/Coxxs/itpcnv-dana/blob/master/lib/uncompress.js#L9) when building .itp)
2. [Take first image from the .itp file.](https://github.com/Coxxs/itpcnv-dana/blob/master/index.js#L66) (The other images are the same images at different resolutions)
3. [Perform a untiling on the image data.](https://github.com/Coxxs/itpcnv-dana/blob/master/index.js#L77)
4. [Add DDS file header to the image data](https://github.com/Coxxs/itpcnv-dana/blob/master/index.js#L77) (BC7 format) for use by other software.

To convert `.dds` back to `.itp`, you may need to reverse these operations, which may not be easy.

AFAIK, the game binary also supports loading old .itp files, maybe you can try using the old .itp generator provided by other projects, but I'm not sure if such resources will display correctly in the game without further modification to game binary.

**Comment by cloudyst** [comment]
thanks for the info,

I'll pass this on to a partner who knows about programming if we can come up with something.

**Comment by Haml0ck** [comment]
[This tool](https://github.com/Kyuuhachi/Aureole/releases) converts PNG to an ITP file that can be used in Ys VIII and Ys IX, and probably a few other Falcom games that use the engine. Just download `cradle-ed7.exe`, drag PNGs to it, and it should convert it to ITP in the same directory. It also converts some ITP formats to PNG (including the ITP it generates), but not the ones from Ys VIII or Ys IX, so keep using itpcnv-dana for that.

Edit: The tool can now convert ITPs from Ys VIII, Ys IX, and Ys Memoire: The Oath in Felghana—so now you have two options.

**Comment by cloudyst** [comment]
> [This tool](https://github.com/Kyuuhachi/Aureole/releases) converts PNG to an ITP file that can be used in Ys VIII and Ys IX, and probably a few other Falcom games that use the engine. Just download `cradle-ed7.exe`, drag PNGs to it, and it should convert it to ITP in the same directory. It also converts some ITP formats to PNG (including the ITP it generates), but not the ones from Ys VIII or Ys IX, so keep using itpcnv-dana for that.

wow!! finally someone did this, impressive!!
Maybe now I'll look into working with Ys9.

**Comment by Kyuuhachi** [comment]
Just put up a new version that can read a couple more itp variants. Can't guarantee it can do all of them, but if you find any files it doesn't support, feel free to send them my way.

**Comment by cloudyst** [comment]
> Just put up a new version that can read a couple more itp variants. Can't guarantee it can do all of them, but if you find any files it doesn't support, feel free to send them my way.

Yes, for ys8 works perfect. And i try some files from ys9 and yes, also works.
